### ISSUE TITLE
Migrate older action format to 2018.2 action syntax

### ISSUE THREAD

**Issue by ebruchez** [orbeon/orbeon-forms#3855]
Following #3803, we now have two separate actions implementations:

- the legacy one, working on XForms-like markup
- the new one, with `<fr:listener>` and `<fr:action>`

Those both translate to XForms, which means we have a duplicate implementation.

This issue is strictly about the runtime. It must convert the old format to the new format first. This means that we will have only one actual implementation of the actions.

We should be able to write this as `<xsl:apply-templates/>` storing the result in a variable, then applying that result further.

This is separate from #1298, which calls for a UI, which means that Form Builder will have to convert from the old format to the new one. That will use the same transformation code, of course.

**Comment by ebruchez** [comment]
- [x] implement the conversion
- [x] make sure `fr:common-dataset-actions-impl` is not lost in translation
    - 2026-04-18 The XSLT function is in `services-and-actions-xf.xsl`. It looks for code in the action syntax using datasets, and generates any missing instances, using an id `fr-dataset-NAME`.

**Comment by ebruchez** [comment]
- [x] when converting, see if  `iterate-control-name` needs to be implemented (#1833)
    - 2026-04-18 See comment below: we decided not to implement it.

**Comment by avernet** [comment]
About `iterate-control-name`:

- In `actions-20182.xsl` we have a comment "TODO: Consider handling `iterate-control-name` per https://github.com/orbeon/orbeon-forms/issues/1833".
- Form Builder does not produce it, confirmed by an open todo on #1833.
- It was only mentioned once in 2014 to an old customer on [Basecamp](https://3.basecamp.com/3600924/buckets/1978678/messages/300962391#__recording_300962542).
- In `actions.xsl`, under `Main event handler to start the action`, we `<xf:action iterate="…">` and then as a sibling copy the other actions like the `xf:send`. This won't result in the service being called "once per row". We broke this in https://github.com/orbeon/orbeon-forms-pe/commit/53a78898f6, done in February 2017 (expand the diff for `actions.xsl` and look for `frf:findRepeatedControlsForTarget`).

Resolution: won't support this; we can always re-add it properly later.

**Comment by avernet** [comment]
Unit tests:

| File                           | How to run                                          | Action syntax | Legacy | Description                                                                                                                             |
|:-------------------------------|:----------------------------------------------------|:--------------|:-------|:----------------------------------------------------------------------------------------------------------------------------------------|
| `tests-actions.xml`            | `formRunnerJVM / testOnly *FormRunnerProcessorTest` | X             |        | Checks `fr:listener` and `fr:action` (action syntax) get compiled into the proper XForms.                                               |
| `ActionsFormat20182Test.scala` | `formRunnerJVM / testOnly *ActionsFormat20182Test`  |  X            |        | Uses `runFormRunner` to run forms using the action syntax. It tests more cases than `tests-actions.xml`, and actually runs the actions. |
| `SimpleActionsTest.scala`      | `formRunnerJVM / testOnly *SimpleActionsTest`       |               | X      | Functionally similar to `ActionsFormat20182Test.scala`, but for the legacy actions.                                                     |
| `DatasetsTest.scala`           | `formRunnerJVM / testOnly *DatasetsTest`            |               | X      | Like `SimpleActionsTest.scala`, this exercises the legacy actions, but focusing on dataset features.                                    |
| `ItemsetActionTest.scala`      | `formRunnerJVM / testOnly *ItemsetActionTest`       |               | X      | Like `SimpleActionsTest.scala`, this exercises the legacy actions, but focusing on itemset features.                                    |

**Comment by avernet** [comment]
Overview of the changes:

- We have a new `actions-legacy-to-20182.xsl`, which we plugged in `unroll-form.xpl` just before `components.xsl`. This way, from the perspective of `components.xsl`, which compiles the action syntax, it is as if the legacy actions didn't exist.
- Most of the code in `actions.xsl` was compiling the legacy actions, and could be removed. We still have some code remaining that dealt with itemsets, instances, submissions, and maybe a few other things. We removed `actions.xsl` and most of the remaining code was moved to `services-and-actions-xf.xsl`.
- Our runtime for the action syntax was in `actions-20182.xsl`. This was renamed to `services-and-actions-fr.xsl`, to reflect that it deals with the `fr:` elements.
- We still have `actions-common.xsl` (unchanged) and `functions.xsl` (mostly unchanged).

In summary, we have 3 main files:

- `actions-legacy-to-20182.xsl` - pre-processing, legacy ⇒ action syntax
- `services-and-actions-xf.xsl` - runtime, handling `xf:` elements
- `services-and-actions-fr.xsl` - runtime, handling `fr:` elements

In `services-and-actions-fr.xsl`, we also made the following changes:

- XBL model support for `fr:listener`
- Support `fr:header`. It was added in January 2026, #6199, as a 2026.1-only feature. It was done in the legacy actions, but not in the action syntax.
